presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanDsl.java

```diff
@@ -0,0 +1,193 @@
+                         */
+                        anyTree(
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testUnreferencedSymbolsDontNeedBinding()
+    {
+        assertPlan("SELECT orderkey, 2 FROM lineitem",
+                output(ImmutableList.of("ORDERKEY"),
+                        anyTree(
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testAliasConstantFromProject()
+    {
+        assertPlan("SELECT orderkey, 2 FROM lineitem",
+                output(ImmutableList.of("ORDERKEY", "TWO"),
+                        project(ImmutableMap.of("TWO", expression("2")),
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testAliasExpressionFromProject()
+    {
+        assertPlan("SELECT 1 + orderkey FROM lineitem",
+                output(ImmutableList.of("EXPRESSION"),
+                        project(ImmutableMap.of("EXPRESSION", expression("1 + ORDERKEY")),
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testTableScan()
+    {
+        assertPlan("SELECT orderkey FROM lineitem",
+                output(ImmutableList.of("ORDERKEY"),
+                        tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey"))));
+    }
+
+    @Test
+    public void testJoinMatcher()
+    {
+        assertPlan("SELECT o.orderkey FROM orders o, lineitem l WHERE l.orderkey = o.orderkey",
+                anyTree(
+                        join(INNER, ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
+                                any(
+                                        tableScan("orders").withAlias("ORDERS_OK", columnReference("orders", "orderkey"))),
+                                anyTree(
+                                        tableScan("lineitem").withAlias("LINEITEM_OK", columnReference("lineitem", "orderkey"))))));
+    }
+
+    @Test
+    public void testAggregation()
+    {
+        assertPlan("SELECT COUNT(nationkey) FROM nation",
+                output(ImmutableList.of("FINAL_COUNT"),
+                        aggregation(ImmutableMap.of("FINAL_COUNT", functionCall("count", ImmutableList.of("PARTIAL_COUNT"))),
+                                any(
+                                        aggregation(ImmutableMap.of("PARTIAL_COUNT", functionCall("count", ImmutableList.of("NATIONKEY"))),
+                                                tableScan("nation", ImmutableMap.of("NATIONKEY", "nationkey")))))));
+    }
+
+    @Test
+    public void testValues()
+    {
+        assertPlan("SELECT * from (VALUES 1, 2)",
+                output(ImmutableList.of("VALUE"),
+                        values(ImmutableMap.of("VALUE", 0))));
```

Comment:
The second argument to values() is the column number that the symbol is bound too. It'd be too much screwing around to try to match the values in the columns for no benefit.